src/textual/widgets/_markdown.py

```diff
@@ -966,7 +966,22 @@ async def forward(self) -> None:
 
     async def _on_markdown_link_clicked(self, message: Markdown.LinkClicked) -> None:
         message.stop()
-        await self.go(message.href)
+        if message.href.find("#") == -1:
+            await self.go(message.href)
+        else:
+            link, anchor = message.href.split("#", 1)
+            await self.go(link)
+
+            def scroll_to_anchor() -> None:
+                toc = self.table_of_contents.table_of_contents
+                assert toc is not None
+                for _, name, block_id in toc:
+                    if name.lower().replace(" ", "-") == anchor:
+                        block = self.query_one(f"#{block_id}", MarkdownBlock)
+                        self.scroll_to_widget(block, top=True)
+                        break
+
+            self.call_later(scroll_to_anchor)
```

Comment:
`call_later` probably isn't the right approach here. It will execute after all other messages have been processed. This is normally very quick, but it is possible that an entirely new document has been loaded in the meantime.

I think this functionality should be built in to `go`, so we can be sure that it applies to the correct document.